tool/preprocess/match.go

```diff
@@ -351,6 +423,10 @@ func (rm *ruleMatcher) match(cmdArgs []string) *rules.RuleBundle {
 					continue
 				}
 			}
+			// Check if all required dependencies are present
```

Comment:
Maybe we can call `rm.matchDependencies` earlier? Maybe skip iterating cmdArgs entirely?